### ISSUE TITLE
unknown address type 206 and 27; add debug

### ISSUE THREAD

**Issue by robtex** [lightningnetwork/lnd#979]
### Background

lnd closes some sessions with `unknown address type: 206` or `unknown address type: 27`
(then it tries to instantly reconnect as https://github.com/lightningnetwork/lnd/issues/941 , eventually resulting in intence SYN to random ports)

there is an issue at clightning regarding this issue. https://github.com/ElementsProject/lightning/issues/1289

their software just relays these without closing the session, with the motivation that it should be possible to add more address types without having to first upgrading *all* nodes in the network. sounds like a good thought. 

regardless how lnd would like to handle new future addresses, the 206 and 27 are clearly wrong, and it would be good to add some debug info in the form of some hexdump or something so we can find the actualy source of these problems.

### Your environment

* version of `lnd`
* which operating system (`uname -a` on *Nix)
* version of `btcd`, `bitcoind`, or other backend
* any other relevant environment details

### Steps to reproduce

Tell us how to reproduce this issue. Please provide stacktraces and links to code in question.

### Expected behaviour

Tell us what should happen

### Actual behaviour

Tell us what happens instead

**Comment by Roasbeef** [comment]
As of this commit (https://github.com/lightningnetwork/lnd/commit/e7d66e1dfd2c73e4fa422ae1a031f56ec7408cea) we don't D/C if we get an unknown address type. The issue is with `clightning` nodes creating and propagating such announcements.

**Comment by robtex** [comment]
awesome!
i've started to see debug-things like this, am i correct to assume this might help the clighting devs? just shooting from the hip thinking 0x19 looks very similar to 27 :)
but yeah, those are genesis blocks, one for testnet and one for mainnet
doesn't look healthy at all, does it?

```2018-03-31 11:49:04.504 [INF] SRVR: Attempting to send msgs 15156 to: 035f460a31af1a03db7965575923735bdfdc96c2ac3e78cec63c9e08c3c9b8a529
2018-03-31 11:49:09.203 [INF] DISC: Broadcasting batch of 33 new announcements
2018-03-31 11:49:24.584 [ERR] DISC: Ignoring ChannelAnnouncement from chain=%v, gossiper on chain=%v 000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943 000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f
2018-03-31 11:49:24.585 [ERR] DISC: Ignoring ChannelAnnouncement from chain=%v, gossiper on chain=%v 000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943 000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f                                                                                                                    
2018-03-31 11:49:24.585 [ERR] DISC: Ignoring ChannelAnnouncement from chain=%v, gossiper on chain=%v 000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943 000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f
```

**Comment by cdecker** [comment]
> The issue is with clightning nodes creating and propagating such announcements.

Forwarding the announcements is correct imho. We might not know how to interpret the address ourselves, but that doesn't mean it's an invalid announcement and that we should drop it. As for c-lightning generating these announcements, I'm still trying to figure out whether it really is a c-lightning node producing them.

**Comment by Roasbeef** [comment]
Agreed the forwarding the is correct, `lnd`. Atm we'll accept them but not forward them as it trigger a halt in our parsing logic at this point.

**Comment by Roasbeef** [comment]
@robtex that log message looks like you're connected to a testnet node from a mainnet node, or the other way around.

**Comment by Roasbeef** [comment]
@cdecker actually, I just realized with the way the spec is written now, we actually _can't_ proceed with parsing. This is due to the fact that we assign a length for each addr type in the spec, instead of having the type _and_ the length follow each other.

**Comment by Roasbeef** [comment]
Fixed by #982.

**Comment by cdecker** [comment]
> @cdecker actually, I just realized with the way the spec is written now, we actually can't proceed with parsing. This is due to the fact that we assign a length for each addr type in the spec, instead of having the type and the length follow each other.

Yep, that's an issue with the way addresses are defined, but the `addrlen` would allow you to continue parsing any appended fields after the `addresses` field.
